test/endpoint_testrunner.py

```diff
@@ -0,0 +1,278 @@
+    :param framework: tensorflow1, tensorflow2, mxnet, pytorch
+    :param version: fw major.minor version, i.e. 2.2
+    :return: default tag python version
+    """
+    if framework == "tensorflow":
+        if Version("2.2") <= Version(version) < Version("2.6"):
+            return "py37"
+        if Version("2.6") <= Version(version) < Version("2.8"):
+            return "py38"
+        if Version(version) >= Version("2.8"):
+            return "py39"
+
+    if framework == "mxnet":
+        if Version(version) == Version("1.8"):
+            return "py37"
+        if Version(version) >= Version("1.9"):
+            return "py38"
+
+    if framework == "pytorch":
+        if Version("1.9") <= Version(version) < Version("1.13"):
+            return "py38"
+        if Version(version) >= Version("1.13") and Version(version) < Version("2.0"):
+            return "py39"
+        if Version(version) >= Version("2.0"):
+            return "py310"
+
+    return "py3"
+
+
+def get_sagemaker_images(images_str, processor):
+    """
+    Gets sagemaker images from images string
+    Assume that all image are of the same framework
+    :param images_str: string containing image uris
+    :param processor: string representing processor type
+    :return: a tuple - framework and a list of image definitions
+    """
+    framework = None 
+    image_definitions = []
+    image_uris = images_str.split(" ")
+    for image_uri in image_uris:
+        if processor in image_uri:
+            if framework is None:
+                framework = utils.get_framework_name(image_uri)
+            py_version = utils.get_py_version(image_uri)
+            framework = utils.get_framework_name(image_uri)
+            framework_version = utils.get_framework_version(image_uri)
+            repository, tag = utils.get_repository_and_tag_from_image_uri(image_uri)
+            registry = utils.get_account_id_from_image_uri(image_uri)
+            region = utils.get_region_from_image_uri(image_uri)
+
+            image_definitions.append({
+                "registry": registry,
+                "region": region,
+                "repository": repository,
+                "framework_version": framework_version,
+                "processor": processor,
+                "py_version": py_version,
+                "tag": tag
+            })
+
+    return framework, image_definitions
+
+
+def get_sagemaker_images_from_github(registry, framework, region, image_type, processor=None):
+    """
+    Return which Inference GPU canary images to run canary tests on for a given framework and AWS region
+
+    :param framework: ML framework (mxnet, tensorflow, pytorch)
+    :param region: AWS region
+    :param image_type: inference
+    :return: a list of space separated image uris
+    """
+    
+    version_regex = {
+        "tensorflow": rf"tf(-sagemaker)-(\d+.\d+)",
+        "mxnet": rf"mx(-sagemaker)-(\d+.\d+)",
+        "pytorch": rf"pt(-sagemaker)-(\d+.\d+)",
+    }
```

Comment:
Please add a TODO to include all SM-DLCs, and not just TF, PT, MX. Or is there a reason to only test these 3 frameworks in SM canaries?

Reply:
At the meantime, the focus is primarily on TF, PT and MX images. The TODO will be added.